frontend/src/pages/NewRunV2.tsx

```diff
@@ -72,11 +80,42 @@ function NewRunV2(props: NewRunV2Props) {
   const [isStartingNewRun, setIsStartingNewRun] = useState(false);
   const [errorMessage, setErrorMessage] = useState('');
   const [isParameterValid, setIsParameterValid] = useState(false);
+  const [clonedRuntimeConfig, setClonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>({});
+  const [usePipelineFromRunLabel, setUsePipelineFromRunLabel] = useState(
+    'Using pipeline from cloned run',
+  );
 
   // TODO(zijianjoy): If creating run from Experiment Page or RunList Page, there is no pipelineId/Version.
   const urlParser = new URLParser(props);
+  const originalRunId = urlParser.get(QUERY_PARAMS.cloneFromRun);
```

Comment:
`NewRunSwitcher` contains this `originalRunId` value. Because we don't know whether the templateString is from pipelineId or runId, I prefer passing `originalRunId` and `pipelineId` via `props`, and also contain a flag to tell whether current templateString is from existing pipeline or existing run.